### ISSUE TITLE
How to use torch.cuda.amp.GradScaler() and autocast to accelerating the training process?

### ISSUE THREAD

**Issue by zzwei1** [ayushi3004/comparative-diffusion-models#1]
Hi. nice work!

I notice you add gradient accumulation to NVIDIA's [DDGAN](https://github.com/NVlabs/denoising-diffusion-gan), I want to add torch.cuda.amp.GradScaler() and autocast to further accelerating the training process. 

Below is an example code for using torch.cuda.amp.GradScaler.

```
scaler = torch.cuda.amp.GradScaler()

for epoch in epochs:
    for input, target in data:
        optimizer0.zero_grad()
        optimizer1.zero_grad()
        with autocast():
            output0 = model0(input)
            output1 = model1(input)
            loss0 = loss_fn(2 * output0 + 3 * output1, target)
            loss1 = loss_fn(3 * output0 - 5 * output1, target)

        # (retain_graph here is unrelated to amp, it's present because in this
        # example, both backward() calls share some sections of graph.)
        scaler.scale(loss0).backward(retain_graph=True)
        scaler.scale(loss1).backward()

        # You can choose which optimizers receive explicit unscaling, if you
        # want to inspect or modify the gradients of the params they own.
        scaler.unscale_(optimizer0)

        scaler.step(optimizer0)
        scaler.step(optimizer1)

        scaler.update()
```
However, I'm not sure about how to add torch.cuda.amp.GradScaler() to the grad_penalty.

https://github.com/ayushi3004/comparative-diffusion-models/blob/b96a7fdd340becf9317ca480c9eb407cb5ffc418/denoising-diffusion-gan/train_ddgan.py#L218
https://github.com/ayushi3004/comparative-diffusion-models/blob/b96a7fdd340becf9317ca480c9eb407cb5ffc418/denoising-diffusion-gan/train_ddgan.py#L228

Could you give me some advice?

Thanks a lot !
